mysql-test/suite/tokudb/t/type_date.test

```diff
@@ -187,16 +187,18 @@ DROP TABLE t1;
 CREATE TABLE t1 (a DATE);
 CREATE TABLE t2 (a DATE);
 CREATE INDEX i ON t1 (a);
---error ER_TRUNCATED_WRONG_VALUE
-INSERT INTO t1 VALUES ('0000-00-00'),('0000-00-00');
---error ER_TRUNCATED_WRONG_VALUE
-INSERT INTO t2 VALUES ('0000-00-00'),('0000-00-00');
+INSERT IGNORE INTO t1 VALUES ('0000-00-00'),('0000-00-00');
+INSERT IGNORE INTO t2 VALUES ('0000-00-00'),('0000-00-00');
+SET sql_mode = 'ALLOW_INVALID_DATES';
```

Comment:
I am not sure this is right.  This is changing the nature of this test sequence.  It seems that we want to see that an error is being kicked back out when we provide an invalid date, this change inverts that and now explicitly allows the invalid dates to be inserted and they later show up in the result for the next two SELECTs.

Reply:
This only changes the mode after the inserts. The block below this is different, there the inserts should succeed. This is how main.type_date looks like now (at least for these two blocks - there are more differences I didn't copy)

Reply:
OK, that's fine then.